internal/controller/template_controller.go

```diff
@@ -344,3 +345,32 @@ func (r *ProviderTemplateReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&hmc.ProviderTemplate{}).
 		Complete(r)
 }
+
+// TemplateSource returns the source of the provided template.
+func TemplateSource(ctx context.Context, c client.Client, tmpl Template) (*sourcev1.HelmRepository, error) {
```

Comment:
- This function is operating on the `Template` interface so it can be used by other controllers as well if needed. That is why I put it in the `template_controller.go` instead of `multiclusterservice_controller.go`.
- Yeah it can be private
- Same reason as I mentioned before about naming getters, but can change it if that's preferable.
- This way is more scalable and correct IMO. I.e. Fetching the source of a template by fetching: Template -> HelmChart -> Source. More correct because Flux's HelmChart and its source must be in the same namespace. More scalable because if we change the name of default repository or if the template exists in a different namespace than "hmc-system" this function will still work.

Reply:
the iface will be removed/reworked soon or later, but it's okay

I have the following concern: currently, all of the templates are being put in the only helm repo, right? if so, then I don't see the reason to make extra API calls, otherwise, surely you're right and there is no other way to properly get the `helmrepository`. we can wait for other opinions though :)

Reply:
Removed the `TemplateSource` function as it was being used only in `helmChartOpts` func anyway.